### ISSUE TITLE
reproduce problem

### ISSUE THREAD

**Issue by ThesecondtoNone** [wwYinYin/SpatialcoGCN#5]
![Image](https://github.com/user-attachments/assets/27f4679e-5d7d-49f1-ae19-63c14c2ef4e2)

  Hello, I'm currently encountering several issues. Firstly, when attempting to reproduce the results in the `simulated_data.ipynb` notebook, the plots generated appear as shown above. Consequently, all subsequent plots are incorrect. Could you please help me understand what might be causing this?
  And Second，When I run the code in files such as run_realdata.ipynb, humanheart.ipynb, and mousebrain.ipynb, I encounter an issue where the data does not exist. May I ask where I can download datasets such as data/realdata/mousebrain, data/realdata/humanheart/Human_heart_ST_gene_count.h5, and data/realdata/mousebrain/20191008080455248897.cnt_data.tsv? Could you please provide guidance on this? Thank you very much!

**Comment by wwYinYin** [comment]
Regarding the first question, your results do not seem to have any problems. Continue to run the contents of simulated_data.ipynb to aggregate your current results into spots. For the second question, humanheart.ipynb and mousebrain.ipynb are just two examples. The data sources have been stated in the article. You can go and see the original article of these data.

**Comment by ThesecondtoNone** [comment]
Thank you for your reply! But is the result of the first question really correct? The coordinate axes in the graph are all incorrect, and when continuing to run the content of simulated_data.ipynb with this result—for example, visualizing the results of simulating spatial transcriptomics data using single-cell data with coordinates—appears even more bizarre, as shown in the figure.

![Image](https://github.com/user-attachments/assets/c19323eb-f7a0-4111-9a31-91b110b42544)

**Comment by wwYinYin** [comment]
1. Regarding the coordinate axis problem of the result of the first figure, you need to manually change
plt.yticks(np.arange(-2000,1400,500))
plt.xticks(np.arange(0,8000,1000))
to get the correct visualization result.
2. Regarding the second figure, you need to customize the height value in
height=80
trans=[[0,height],[int(height*math.sqrt(3)/2),height/2],[int(height*math.sqrt(3)/2),-height/2],
[0,-height],[-int(height*math.sqrt(3)/2),-height/2],[-int(height*math.sqrt(3)/2),height/2]]
according to the coordinate axis and your own needs

**Comment by ThesecondtoNone** [comment]
And While I am running the run_impute.ipynb, I encountered some data content, such as data/DataUpload/Dataset2(9) and:
Spatial_file = DataDir + 'Insitu_count.txt'
location_file = DataDir + 'Locations.txt'
RNA_data_adata = sc.read_h5ad(DataDir + 'starmap_sc_rna.h5ad')，Where can I download these data? Could you please provide detailed addresses or data resources? Thank you!

#DataDir="./data/realdata/humanheart/results/"
Spatial_file = DataDir + 'filtered_matrix1_transposed.txt'
location_file = DataDir + 'fixed_coordinates.csv'

RNA_data_adata = sc.read_h5ad(DataDir+'starmap_sc_rna.h5ad')
Spatial_data_adata = sc.read(Spatial_file, sep = '\t')
spatial_loc = pd.read_csv(location_file,sep='\t')
train_gene = np.load(DataDir + 'train_list.npy', allow_pickle = True).tolist()
predict_gene = np.load(DataDir + 'test_list.npy', allow_pickle = True).tolist()
Spatial_data_adata.obs=spatial_loc

RNA_ret_adata=RNA_data_adata.copy()
sc.pp.normalize_total(RNA_ret_adata)
RNA_ret_adata=adata_to_cluster_expression(RNA_ret_adata)
print(RNA_data_adata)#
It's the content in the code you provided, and could you also provide train_list.npy and test_list.npy? Thank you very much!

**Comment by ThesecondtoNone** [comment]
![Image](https://github.com/user-attachments/assets/22e13aa6-fa52-4a50-941e-d2804bcd3873)

![Image](https://github.com/user-attachments/assets/c91b3c7c-fd20-4637-b4a9-df8b4b14332c)Thank you very much for your prompt reply! Regarding Dataset 2 used in your tutorial's run_impute.ipynb file, I searched according to Appendix 3, but here I am confused that there is no downloadable ST data available. What is the situation? The MERFISH dataset only contains MERFISH_genes.csv, lacking the Insitu_count.txt, Locations.txt, train_list.npy, test_list.npy, and other files required in the tutorial code, making it impossible to successfully run and reproduce the run_impute.ipynb file.

**Comment by wwYinYin** [comment]
Dataset 2 and other data in run_impute.ipynb can be found at https://figshare.com/articles/dataset/SpatialcoGCN_data/22682611. The starmap_sc_rna.h5ad mentioned in run_impute.ipynb needs to be replaced with scRNA.h5ad in the corresponding folder. trian_list.npy and test_list.npy represent the training genes and test gene sets, which can be defined by yourself, for example, the first 80% of genes are used as training genes, and the last 20% of genes are used as test genes. In addition, the source of other data can also refer to the DATA AVAILABILITY section of our article. We have no plans to disclose other preprocessed data for the time being.

**Comment by ThesecondtoNone** [comment]
![Image](https://github.com/user-attachments/assets/66884472-2d55-44cc-bae6-c50beb6f6db7)Thank you for your follow-up guidance! As per your instructions, when running the run_impute.ipynb file using dataset9 from SpatialcoGCN-SimData, I encountered the issue shown in the screenshot. Could this problem stem from the concat_st_sc step, resulting in NaN values during the subsequent ad_map training and a persistently stagnant training loss of 2.0? Could you kindly assist me in diagnosing and resolving this? Your help would be greatly appreciated!

**Comment by ThesecondtoNone** [comment]
Hello, I am using dataset9 from SpatialcoGCN-SimData as the input data, because I found that its genes best match the 250 genes specified in your paper — with 225 used as the training set and 25 as the test set. However, when running the run_impute.ipynb file, I encountered the issue mentioned above. What do you think might be the cause? I look forward to your reply! Thank you!
